include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -301,6 +301,14 @@
 #define _ONEDPL_CPP20_SHIFT_LEFT_RIGHT_PRESENT                                                                         \
     (_ONEDPL___cplusplus >= 202002L && (_MSC_VER >= 1921 || _GLIBCXX_RELEASE >= 10))
 
+#if defined(_MSVC_LANG) && _MSVC_LANG >= 202002L
+#   define _ONEDPL_CPP20_RANGES_PRESENT (__cpp_lib_ranges >= 201911L)
```

Comment:
Should `_ONEDPL_CPP20_RANGES_PRESENT` be a public feature macro? It might have different values depending on a set of available algorithms.

Reply:
I am not sure that `_ONEDPL_CPP20_RANGES_PRESENT` should be public.. it macro is defined by C++ library version and C++ version.
A subset of implemented algorithms in `oneapi::dpl::ext::ranges::` oneDPL is (will be) defined via oneDPL version, I guess.

Reply:
Let's then avoid using `_ONEDPL_CPP20_RANGES_PRESENT` in the tests, since it is a private macro

Reply:
I think the original question has been answered. The question about how to define `_ONED_L_CPP20_RANGES_PRESENT` can be further discussed in https://github.com/oneapi-src/oneDPL/pull/1457/files#r1691478417.